include/boost/geometry/iterators/detail/segment_iterator/range_segment_iterator.hpp

```diff
@@ -100,23 +100,40 @@ class range_segment_iterator
             Reference
         >
 {
+private:
+    static bool has_less_than_two_elements(Range const& r)
+    {
+        iterator_type first = range_iterator_begin<Range>::apply(r);
+        iterator_type last = range_iterator_end<Range>::apply(r);
+
+        return first == last || ++first == last;
+    }
+
 public:
     typedef typename range_iterator_type<Range>::type iterator_type;
 
     // default constructor
     range_segment_iterator()
-        : m_it()
+        : m_it(), m_has_less_than_two_elements()
```

Comment:
@barendgehrels Why it can't be assumed that a class may not have a default ctor? It's a common thing! It may have any number of non-default ctors taking arbitrary types. I don't understand this reasoning.

Furthermore we shouldn't assume that a default-initialization is zero-initialization.

What is more, we already require that T must be initialized explicitly in other places. In your fix above (assignment instead of an explicit ctor call) it makes no difference because a ctor will be called (at some point on a path of implicit conversions). E.g. here:

    return_type const zero = 0; // implicit conversion <=> ctor

The library must and do require that there must be some path of implicit conversions from double/int to T. Effectively it's requiring ctor taking double or some other type constructible from numeric constant. So why add another requirement (zero-initializing default ctor)?

And why value-initialization is better than explicit construction? It's worse, because with explicit construction you see immediately what's passed, exactly like with `false` or `NULL`.

Or maybe I'm missing something?